casual/casual-jca/src/main/java/se/laz/casual/jca/discovery/CasualDiscoveryCaller.java

```diff
@@ -47,11 +47,11 @@ public static CasualDiscoveryCaller of(CasualManagedConnection managedConnection
     public DiscoveryReturn discover(UUID corrid, List<String> serviceNames, List<String> queueNames)
     {
         LOG.finest(() -> "issuing domain discovery, corrid: " + PrettyPrinter.casualStringify(corrid) + " service names: " + serviceNames + " queue names: " + queueNames);
-        Domain domain = ConfigurationService.getInstance().getConfiguration().getDomain();
```

Comment:
If it was useful it would have been used as a parameter into a method somewhere, but it is not, anywhere, at all. Equally DomainId, though i have used that, it does not really provide any value add. If you go too far down this route you would have a class for every field, HostName, PortNumber, ServiceName, QueueName. You have to be able to use primatives at some point.
The only places Domain was used was to retrieve it from the json, which in turn didn't use a DomainId, because it can't becase that is not how the json structure is provided as i mentioned earlier. The usages of Domain object always immediately just split the object up into it's string and UUID components to send into other classes which required String and UUID.

Reply:
Sure, we could go with primitives everywhere - that would be horrible though.
Speaking of DomainId, it is also used in casual-caller, as DomainId - it does not have to know what kind of stuff is inside of it.
As for Domain, it is a mental construct as well, a java Domain from our point of view consists of a name ( should probably be a default and not empty if not set) and a unique DomainId.

Reply:
yes, but as I am saying, it can go too far the other way too.
Equally the "domain" concept for casual is already well used and means more than just those two things.
Anyway, I think we have entered bike shed territory here, so lets just park this.